api.go

```diff
@@ -909,6 +909,23 @@ func (lapi *Client) Heartbeat(id string) error {
 	return err
 }
 
+// LockPull locks the stream pull
+func (lapi *Client) LockPull(id string, leaseTimeoutMs time.Duration) error {
+	if id == "" {
+		return errors.New("empty id")
+	}
+
+	payload := struct {
+		leaseTimeout time.Duration
+	}{
+		leaseTimeout: leaseTimeoutMs,
+	}
```

Comment:
**nit:** since you're dynamically typing it here you could also just use a raw JSON object and avoid the weirdish repetition
```suggestion
	payload := map[string]interface{}{
		"leaseTimeout": leaseTimeoutMs,
	}
```
(could be `map[string]time.Duration` in this case as well)